src/index.ts

```diff
@@ -68,56 +70,53 @@ const conditionMatch = (
 
 // Support more complex fuzzy matching rules,
 // backward compatible with the definition of the specification
-const fuzzyMatchKey = (path: string, keys: Array<string>) => {
-  let prefix;
-  let matched;
-  const data = [];
+const fuzzyMatchKey = (path: string, obj: Imports) => {
   const pathLen = path.length;
+  const keys = Object.keys(obj).sort((a, b) => b.length - a.length);
+  const data = [];
 
-  const findPathMatchIdx = (char: string | undefined, idx: number) => {
-    if (!char) return pathLen;
-    for (let i = idx; i < pathLen; i++) {
-      if (char === path[i]) return i;
-    }
-    return -1;
-  };
-
-  keys = keys.sort((a, b) => b.length - a.length);
+  let prefix;
+  let matched;
 
   for (const key of keys) {
     if (matched) break;
+
+    // path index
     let i = 0;
+
+    // key index
     let j = 0;
+
     for (i = 0; i < pathLen; i++) {
       if (path[i] === key[j]) {
         j++;
       } else if (key[j] === '*') {
         const next = key[j + 1];
+        // '**' is invalid
         if (next === '*') break;
-        const pathMatchIdx = findPathMatchIdx(next, i + 1);
-        if (pathMatchIdx === -1) break;
-        data.push(path.slice(i, pathMatchIdx));
+        // get index of next char
+        const k = !next ? pathLen : path.indexOf(next, i + 1);
+        // exit early
+        if (k === -1) break;
```

Comment:
Maybe another guard check is required to resolve subpath mappings which have extensions
e.g. './foo/*.js' => './foo/*.ts'